Bukkit/src/main/java/me/egg82/tcpp/commands/TrollCommand.java

```diff
@@ -102,7 +103,109 @@ public void onAttach(CommandIssuer issuer, String command) {
+    @CommandPermission("tcpp.command.garble")
+    @Description("{@@garble.description}")
+    @Syntax("<player>")
+    @CommandCompletion("@player")
+    public void onGarble(CommandIssuer issuer, String player) {
+        TrollType type = TrollType.GARBLE;
+        getChain(issuer, player).syncLast(v -> startOrStopTroll(issuer, v, type, true, plugin, v, type)).execute();
+    }
+
+    @Subcommand("infinity")
+    @CommandPermission("tcpp.command.infinity")
+    @Description("{@@infinity.description}")
+    @Syntax("<player>")
+    @CommandCompletion("@player")
+    public void onInfinity(CommandIssuer issuer, String player) {
+        TrollType type = TrollType.INFINITY;
+        getChain(issuer, player).syncLast(v -> startOrStopTroll(issuer, v, type, true, plugin, v, type)).execute();
+    }
+
+    @Subcommand("levitate")
+    @CommandPermission("tcpp.command.levitate")
+    @Description("{@@levitate.description}")
+    @Syntax("<player>")
+    @CommandCompletion("@player")
+    public void onLevitate(CommandIssuer issuer, String player) {
+        TrollType type = TrollType.LEVITATE;
+        getChain(issuer, player).syncLast(v -> startOrStopTroll(issuer, v, type, true, plugin, v, type)).execute();
+    }
+
+    @Subcommand("lift")
+    @CommandPermission("tcpp.command.lift")
+    @Description("{@@lift.description}")
+    @Syntax("<player>")
+    @CommandCompletion("@player")
+    public void onLift(CommandIssuer issuer, String player) {
+        TrollType type = TrollType.LIFT;
+        getChain(issuer, player).syncLast(v -> startTroll(issuer, v, type, true, v, type)).execute();
+    }
+
+    @Subcommand("lock")
+    @CommandPermission("tcpp.command.lock")
+    @Description("{@@lock.description}")
+    @Syntax("<player>")
+    @CommandCompletion("@player")
+    public void onLock(CommandIssuer issuer, String player) {
+        TrollType type = TrollType.LOCK;
+        getChain(issuer, player).syncLast(v -> startOrStopTroll(issuer, v, type, true, plugin, v, type)).execute();
+    }
+
+    @Subcommand("snowballfight")
+    @CommandPermission("tcpp.command.snowballfight")
+    @Description("{@@snowballfight.description}")
+    @Syntax("<player>")
+    @CommandCompletion("@player")
+    public void onSnowballFight(CommandIssuer issuer, String player) {
+        TrollType type = TrollType.SNOWBALLFIGHT;
+        getChain(issuer, player).syncLast(v -> startOrStopTroll(issuer, v, type, true, plugin, v, type)).execute();
+    }
+
+    @Subcommand("swap")
+    @CommandPermission("tcpp.command.swap")
+    @Description("{@@swap.description}")
+    @Syntax("<player> <player>")
+    @CommandCompletion("@player @player")
+    public void onSwap(CommandIssuer issuer, String player1, String player2) {
+        TrollType type = TrollType.SWAP;
+
+        Player p1 = Bukkit.getPlayer(player1);
+        Player p2 = Bukkit.getPlayer(player2);
+        if (p1 == null) {
+            issuer.sendError(Message.ERROR__PLAYER_NOT_FOUND, "{player}", player1);
```

Comment:
Create a new getChain method which takes two player args instead of one. Much cleaner.